### PR TITLE
Replacing an array with 'performDiff: false' not replacing the array.

### PR DESCRIPTION
**Pull request by AnthonyMDev**
This has been tested in my application and works as expected.

I'm not sure if this is the approach that you want to take, but I figured I'd make a PR anyway. It's a pretty simple fix and the most elegant way I came up with.

### CODE REVIEW

File: Sources/ObservableArray.swift [hunk 1]
```diff
@@ -315,6 +316,9 @@ extension MutableObservableArray where Item: Equatable {
         subject.next(ObservableArrayEvent(change: .deletes(deletes), source: self))
         subject.next(ObservableArrayEvent(change: .inserts(inserts), source: self))
         subject.next(ObservableArrayEvent(change: .endBatchEditing, source: self))
+      } else {
+        self.array = array
+        subject.next(ObservableArrayEvent(change: .replace, source: self))
```
Comment (srdanrasic) [inline Sources/ObservableArray.swift]: Can we put these two lines inside of a `lock.atomic` block?

**Comment by AnthonyMDev** [comment]
You got it. I'll handle this when I get back to my workstation.

File: Sources/ObservableArray.swift (hunk 1 above)
Comment (AnthonyMDev) [inline Sources/ObservableArray.swift]: Actually if you look up about it's still wrapped in the `lock.atomic.`

The entire `if performDiff { ... } else { ... }` is wrapped in the `lock.atomic.

**Comment by AnthonyMDev** [comment]
Also, should we have a `replace()` function that does not have a `performDiff` parameter that is performable on a `MutableObservableArray` that does not conform to `Equatable`?

I think that is an important piece of functionality. Can I make a separate PR for that ?

File: Sources/ObservableArray.swift (hunk 1 above)
Comment (srdanrasic) [inline Sources/ObservableArray.swift]: Ah, yes, you are right :)

**Comment by srdanrasic** [comment]
Makes sense! Sure, feel free to do that :)

**Comment by AnthonyMDev** [comment]
Thanks for the merge. I'll handle the other PR later this afternoon.

**Comment by Joebayld** [comment]
Hey guys. Should the new replace() function be public? Right now it's not accessible. That way it's possible to do a replace on the MutableObservableArray without the items being Equatable

**Comment by AnthonyMDev** [comment]
That's my fault! I'll fix this in the morning. Good catch!

**Pull request merged**